Cython/Compiler/ModuleNode.py

```diff
@@ -2542,151 +2516,178 @@ def generate_import_star(self, env, code):
         code.putln(UtilityCode.load_as_string("ImportStar", "ImportExport.c")[1])
         code.exit_cfunc_scope()  # done with labels
 
-    def generate_module_state_start(self, env, code):
-        # TODO: Refactor to move module state struct decl closer to the static decl
-        code.putln("#if CYTHON_USE_MODULE_STATE")
-        code.putln('typedef struct {')
-        code.putln('PyObject *%s;' % Naming.builtins_cname)
-        code.putln('PyObject *%s;' % Naming.cython_runtime_cname)
-        code.putln('PyObject *%s;' % Naming.empty_tuple)
-        code.putln('PyObject *%s;' % Naming.empty_bytes)
-        code.putln('PyObject *%s;' % Naming.empty_unicode)
+    def generate_module_state_begin(self, env, code):
+        code.putln('static int %s;' % Naming.lineno_cname)
+        code.putln('static int %s = 0;' % Naming.clineno_cname)
+        code.putln('static const char * %s = %s;' % (Naming.cfilenm_cname, Naming.file_c_macro))
+        code.putln('static const char *%s;' % Naming.filename_cname)
```

Comment:
I actually hope that we're not using these any more. They should be function local now (again). But it's not part of this PR to try if they can be removed.